config/database.yml

```
@@ -0,0 +1,94 @@
+# PostgreSQL. Versions 9.3 and up are supported.
+#
+# Install the pg driver:
+#   gem install pg
+# On macOS with Homebrew:
+#   gem install pg -- --with-pg-config=/usr/local/bin/pg_config
+# On macOS with MacPorts:
+#   gem install pg -- --with-pg-config=/opt/local/lib/postgresql84/bin/pg_config
+# On Windows:
+#   gem install pg
+#       Choose the win32 build.
+#       Install PostgreSQL and put its /bin directory on your path.
+#
+# Configure Using Gemfile
+# gem "pg"
+#
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  # For details on connection pooling, see Rails configuration guide
+  # https://guides.rubyonrails.org/configuring.html#database-pooling
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  timeout: 5000
+  # GitHub Actions上でのCI実行時には、コンテナ上で動作しているDBに対して、ローカルホストのポートからアクセスする。
+  # https://docs.github.com/en/actions/using-containerized-services/about-service-containers#running-jobs-on-the-runner-machine
+  #
+  # GitHub Actions上でのCI実行時には、環境変数 CI=true が常に設定されている。
+  # https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+  host: <%= ENV['CI'] == 'true' ? 'localhost' : 'db' %>
```

この手のものをハードコーディングすると、環境が増えたときに厄介なので、環境変数で差し込めるようにするのがよいですね。
```suggestion
  host: <%= ENV['POSTGRES_HOST'] %>
```

環境変数を省略した場合にデフォルト値を与えたい、って場合はこう書くかな。
```suggestion
  host: <%= ENV.fetch('POSTGRES_HOST', 'localhost') %>
```

---

確かに単純に環境変数経由で値を設定するようにしたほうがシンプルでいいですね。修正します。

---

実際のプロジェクトでは、いろんな環境を用意することになるので、外から指定できるようになっていると便利です

* ローカルの開発環境
* 本番環境
* その他の環境 (ステージング環境、QA環境などなど)